src/components/documents/CmsText/index.tsx

```diff
@@ -0,0 +1,31 @@
+import { observer } from 'mobx-react-lite';
+import { CmsTextContext, useFirstCmsTextDocumentIfExists } from '@tdev-components/documents/CmsText/shared';
+import React from 'react';
+
+interface Props {
+    id?: string;
+}
+
+const CmsText = observer(({ id }: Props) => {
+    const context = React.useContext(CmsTextContext);
```

Comment:
Then we could do here:
```tsx
const contextId =  React.useContext(CmsTextContext)?.id;
const cmsText = useFirstCmsTextDocumentIfExists(id || contextId);
...
```

Additional advantage: this for sure doesent lead to a`Uncaught Invariant Violation: Rendered more hooks than during the previous render.` errors. This could, only in theory, happen in the code below. But i needed to analyze the code much more in depth to be sure, it should work without issues...